src/osd/PrimaryLogPG.cc

```diff
@@ -3457,24 +3458,121 @@ void PrimaryLogPG::cancel_manifest_ops(bool requeue, vector<ceph_tid_t> *tids)
   }
 }
 
-void PrimaryLogPG::refcount_manifest(ObjectContextRef obc, object_locator_t oloc, hobject_t soid,
-                                     SnapContext snapc, bool get, RefCountCallback *cb, uint64_t offset)
+void PrimaryLogPG::dec_refcount(ObjectContextRef obc, const object_info_t& oi, 
+				const object_ref_delta_t& refs)
+{
+  for (auto p = refs.begin(); p != refs.end(); ++p) {
+    dout(10) << __func__ << ": decrement reference on offset oid: " << p->first << dendl;
+    object_locator_t target_oloc(p->first);
+    refcount_manifest(obc, obc->obs.oi.soid, target_oloc, p->first, 
+		      SnapContext(), refcount_t::DECREMENT_REF, NULL);
+  }
+}
+
+
+void PrimaryLogPG::dec_all_refcount_head_manifest(object_info_t& oi, OpContext* ctx, const hobject_t &coid)
```

Comment:
This now gets called on clones, so this is not a good name.

Reply:
I don't think it makes sense to pass both oi and coid.  In trim_object, I think that choice caused you to pass the wrong object_info_t  and in every case where this method is called on head, coid is hobject_t() with misleading values for snapid.